src/error.ts

```diff
@@ -1255,9 +1255,8 @@ export function needsRetryableWriteLabel(
       // use original top-level code from server response
       return RETRYABLE_WRITE_ERROR_CODES.has(error.result.code ?? 0);
     }
-    return RETRYABLE_WRITE_ERROR_CODES.has(
-      error.result.writeConcernError.code ?? Number(error.code) ?? 0
-    );
+    const code = error.result.writeConcernError.code ?? Number(error.code);
```

Comment:
eslint correctly points out that `Number(...)` never returns null/undefined, so the nullish coelescing was unused.  I believe what we want instead is a Number.isNumber check.